src/libs/CategoryOptionListUtils.ts

```diff
@@ -0,0 +1,282 @@
+    const sortedCategories = sortCategories(categories);
+    const enabledCategories = Object.values(sortedCategories).filter((category) => category.enabled);
+    const enabledCategoriesNames = enabledCategories.map((category) => category.name);
+    const selectedOptionsWithDisabledState: Category[] = [];
+    const categorySections: CategoryTreeSection[] = [];
+    const numberOfEnabledCategories = enabledCategories.length;
+
+    selectedOptions.forEach((option) => {
+        if (enabledCategoriesNames.includes(option.name)) {
+            const categoryObj = enabledCategories.find((category) => category.name === option.name);
+            selectedOptionsWithDisabledState.push({...(categoryObj ?? option), isSelected: true, enabled: true});
+            return;
+        }
+        selectedOptionsWithDisabledState.push({...option, isSelected: true, enabled: false});
+    });
+
+    if (numberOfEnabledCategories === 0 && selectedOptions.length > 0) {
+        const data = getCategoryOptionTree(selectedOptionsWithDisabledState, true);
+        categorySections.push({
+            // "Selected" section
+            title: '',
+            shouldShow: false,
+            data,
+            indexOffset: data.length,
+        });
+
+        return categorySections;
+    }
+
+    if (searchValue) {
+        const categoriesForSearch = [...selectedOptionsWithDisabledState, ...enabledCategories];
+        const searchCategories: Category[] = [];
+
+        categoriesForSearch.forEach((category) => {
+            if (!category.name.toLowerCase().includes(searchValue.toLowerCase())) {
+                return;
+            }
+            searchCategories.push({
+                ...category,
+                isSelected: selectedOptions.some((selectedOption) => selectedOption.name === category.name),
+            });
+        });
+
+        const data = getCategoryOptionTree(searchCategories, true);
+        categorySections.push({
+            // "Search" section
+            title: '',
+            shouldShow: true,
+            data,
+            indexOffset: data.length,
+        });
+
+        return categorySections;
+    }
+
+    if (selectedOptions.length > 0) {
+        const data = getCategoryOptionTree(selectedOptionsWithDisabledState, true);
+        categorySections.push({
+            // "Selected" section
+            title: '',
+            shouldShow: false,
+            data,
+            indexOffset: data.length,
+        });
+    }
+
+    const selectedOptionNames = selectedOptions.map((selectedOption) => selectedOption.name);
+    const filteredCategories = enabledCategories.filter((category) => !selectedOptionNames.includes(category.name));
+
+    if (numberOfEnabledCategories < CONST.CATEGORY_LIST_THRESHOLD) {
+        const data = getCategoryOptionTree(filteredCategories, false, selectedOptionsWithDisabledState);
+        categorySections.push({
+            // "All" section when items amount less than the threshold
```

Comment:
Could you explain this comment?

Reply:
Oh I see, these are the sections comments- maybe instead of `const data` we should use the variable name to make this clearer

Reply:
hm agree, this is just copied over code from how it was previously used. Do we need to make this change in this PR?